### PR TITLE
Assignment week 2 concurrency, channel

### PR DESCRIPTION
**Pull request by nguyenhuuluan434**
Hi,
This is my PR. Please help me review.
Thanks
Luân Nguyễn Hữu

### CODE REVIEW

**Comment by txchuyen** [comment]
Thank you for submitting and happy coding 👏

How about if not using wait group?

File: week_2/Assignment/AtHome/Error.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package main
+
+type customError struct {
```
Comment (txchuyen) [inline week_2/Assignment/AtHome/Error.go]: It's better if we specify the error to indicate what error is, examples:
- type ErrFileNotFound struct{...}
- type ErrFileAccess struct{...}

File: week_2/Assignment/AtHome/fileHandle.go [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+
+const maxLengthLine = 200
+const maxBufferSize = 10000
+
+type FileInfo struct {
+	FileName string
+	Path     string
+}
+
+type S3FileInfo struct {
```
Comment (txchuyen) [inline week_2/Assignment/AtHome/fileHandle.go]: It's better if we split this into 3 files
- reader.go
- s3.go
- file.go

File: week_2/Assignment/AtHome/fileHandle.go [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+	var result = make(map[string]int)
+	for _, word := range words {
+		result[word] += 1
+	}
+	output <- result
+	close(output)
+}
+
+
+func processor(paths []string, wg *sync.WaitGroup) (result map[string]int, err error) {
```
Comment (txchuyen) [inline week_2/Assignment/AtHome/fileHandle.go]: the function name should be a verb

**Review by txchuyen — COMMENTED** [review]
Thank you for submitting and happy coding 👏

**Comment by nguyenhuuluan434** [comment]
Thank you for taking your time for review my code.

**Pull request merged**